homeassistant/components/tibber/diagnostics.py

```diff
@@ -0,0 +1,39 @@
+"""Diagnostics support for Tibber."""
+from __future__ import annotations
+
+import tibber
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ACCESS_TOKEN
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+
+REDACTED = "REDACTED"
+
+
+async def async_get_config_entry_diagnostics(
+    hass: HomeAssistant, config_entry: ConfigEntry
+) -> dict:
+    """Return diagnostics for a config entry."""
+    tibber_connection: tibber.Tibber = hass.data[DOMAIN]
+
+    diagnostics_data = {
+        "info": {
+            **config_entry.as_dict(),
+        },
```

Comment:
No need to copy the dict or store whole config entry.
```suggestion
        "info": config_entry.data,
```

Reply:
Actually, redact the token:
```suggestion
        "info": {
            **config_entry.data,
            CONF_ACCESS_TOKEN: REDACTED
        },
```